pages/header.py

```diff
@@ -36,16 +39,37 @@ def sign_out_button(self, username, password):
         else:
             self.click(self.const.SIGN_OUT_BUTTON_XPATH)
 
-    # @log_wrapper
     @log_wrapper
     def navigate_to_create_post(self):
         """Navigate to create post page via header Create Post button"""
         self.click(xpath=self.const.CREATE_POST_BUTTON_XPATH)
 
     @log_wrapper
     def navigate_to_start_page(self):
-        """Navigate to Start Page page via header Sign Out button"""
+        """Navigate to Start Page via header Sign Out button"""
         self.click(xpath=self.const.SIGN_OUT_BUTTON_XPATH)
 
-        from pages.create_post_page import CreatePostPage
-        return CreatePostPage(self.driver)
+        from pages.start_page import StartPage
+        return StartPage(self.driver)
+
+    @wait_until_ok(timeout=3, period=0.5)
+    def click_and_validate_my_profile_button(self):
+        """Click on My Profile button until it disappear"""
+        self.click(self.const.MY_PROFILE_BUTTON_XPATH)
+        assert not self.is_element_exists(self.const.MY_PROFILE_BUTTON_XPATH), "My Profile button didn't disappear"
+
+    @log_wrapper
+    def my_profile_button(self, username, password):
```

Comment:
Do you really need such validation for this button ?

Reply:
Fixed: validation method for My Profile button was deleted